AVsitter2/Utilities/Missing-anim-finder.lsl

```diff
@@ -67,29 +67,36 @@ default
             if (data == EOF)
             { 
                 if(!NOT_FOUND_COUNT){
-                    Owner_Say("All animations referenced in the notecard were accounted for.");   
+                    Owner_Say("All anims referenced in the notecard were accounted for.");   
                 }
                 else{
                     Owner_Say("Anims were used in the notecard but not found in inventory!");
                 }
                 integer i;
                 for (i = 0; i < llGetInventoryNumber(INVENTORY_ANIMATION); i++)
                 {
+                    integer index;
                     integer isVariableSpeed;
                     string anim_basename = llGetInventoryName(INVENTORY_ANIMATION, i);
-                    if(llListFindList(["+","-"], [llGetSubString(anim_basename,-1,-1)]) != -1)
+                    
+                    if(llListFindList(["+","-"], [llDeleteSubString(anim_basename,0,-2)]) != -1)
```

Comment:
Mixing negative and positive indexes (except 0 and -1) should always raise alarms because of its corner case behaviours, which are undesirable in most circumstances, and only by chance happen to be desirable here.

In this case the code would be correct either way: it's not possible to enter an animation with an empty name, as it will be filtered out by `llParseString2List` while reading the notecard, therefore there's no way that an empty string can possibly pass the `index!=-1` test a few lines below because it will never be in `ALL_USED_ANIMS`. Consequently, this whole `if` doesn't matter for animations named `+` or `-`, even when correctly taking the last character of the string.

The only gain is therefore a tiny performance improvement in case an animation with a name of `+` or `-` is found, because it's stopped earlier. I think it's not worth having to stop and think whether the actual behaviour in the corner cases is correct while reading the code, regardless of it being correct.

So, please change it to `llGetSubString(anim_basename, -1, -1)`.

Reply:
Should be fixed now :)